clio-integration-test/src/it/scala/org/broadinstitute/clio/integrationtest/tests/UbamTests.scala

```diff
@@ -52,113 +50,35 @@ trait UbamTests { self: BaseIntegrationSpec =>
-    val tmpMetadata =
-      writeLocalTmpJson(UbamMetadata(project = Some("fake_project")))
-
-    recoverToSucceededIf[RuntimeException] {
-      runIgnore(
-        ClioCommand.addHybselUbamName,
-        "--flowcell-barcode",
-        stubKey.flowcellBarcode,
-        "--lane",
-        stubKey.lane.toString,
-        "--library-name",
-        stubKey.libraryName,
-        "--location",
-        stubKey.location.entryName,
-        "--metadata-location",
-        tmpMetadata.toString
-      )
-    }
-  }
-
-  it should "throw a FailedResponse 404 when running query command for hybsel ubams" in {
-    val queryResponseFuture = runCollectJson(
-      ClioCommand.queryHybselUbamName,
-      "--flowcell-barcode",
-      stubKey.flowcellBarcode
-    )
-    recoverToExceptionIf[FailedResponse](queryResponseFuture)
-      .map(statusCodeShouldBe(StatusCodes.NotFound))
-  }
-
-  def messageShouldBe(expectedMessage: String): Exception => Assertion = { e: Exception =>
-    e.getMessage should be(expectedMessage)
-  }
-
-  it should "throw a FailedResponse 404 when running move command for hybsel ubams" in {
-    val moveResponseFuture = runDecode[UpsertId](
-      ClioCommand.moveHybselUbamName,
-      "--flowcell-barcode",
-      stubKey.flowcellBarcode,
-      "--lane",
-      stubKey.lane.toString,
-      "--library-name",
-      stubKey.libraryName,
-      "--location",
-      stubKey.location.entryName,
-      "--destination",
-      "gs://fake-path/"
-    )
-    recoverToExceptionIf[FailedResponse](moveResponseFuture)
-      .map(statusCodeShouldBe(StatusCodes.NotFound))
-  }
-
-  it should "throw a FailedResponse 404 when running delete command for hybsel ubams" in {
-    val deleteResponseFuture = runDecode[UpsertId](
-      ClioCommand.deleteHybselUbamName,
-      "--flowcell-barcode",
-      stubKey.flowcellBarcode,
-      "--lane",
-      stubKey.lane.toString,
-      "--library-name",
-      stubKey.libraryName,
-      "--location",
-      stubKey.location.entryName,
-      "--note",
-      "note"
-    )
-    recoverToExceptionIf[FailedResponse](deleteResponseFuture)
-      .map(statusCodeShouldBe(StatusCodes.NotFound))
-  }
-
-  it should "create the expected wgs-ubam mapping in elasticsearch" in {
+  it should "create the expected  ubam mapping in elasticsearch" in {
     import com.sksamuel.elastic4s.http.ElasticDsl._
     import ElasticsearchUtil.HttpClientOps
 
-    val expected = ElasticsearchIndex.WgsUbam
+    val expected = ElasticsearchIndex.Ubam
     val getRequest = getMapping(IndexAndType(expected.indexName, expected.indexType))
 
     elasticsearchClient.executeAndUnpack(getRequest).map {
       _ should be(Seq(indexToMapping(expected)))
     }
   }
 
-  // Generate a test for every possible Location value.
-  Location.values.foreach {
-    it should behave like testWgsUbamLocation(_)
+  // Generate a test for every combination of Location value and command type.
+  Location.values.foreach { location =>
```

Comment:
I think this could be phrased as a for-comprehension:
```
for {
    location <- Location.values,
    useNewCommands <- Seq(true, false)
} yield { it should behave like testUbamLocation(location, useNewCommands) }```
and then you don't have to nest.